src/internal/miscutil/miscutil.go

```diff
@@ -16,9 +16,7 @@ func WithPipe(wcb func(w io.Writer) error, rcb func(r io.Reader) error) error {
 	pr, pw := io.Pipe()
 	eg := errgroup.Group{}
 	eg.Go(func() error {
-		err := wcb(pw)
-		pw.CloseWithError(err)
-		return errors.EnsureStack(err)
+		return errors.EnsureStack(pw.CloseWithError(wcb(pw)))
```

Comment:
I'm not 100% convinced that PipeWriter.CloseWithError returns the error that you pass to it.  I think it might return nil actually.  (see src/io/pipe.go.)

Reply:
Omg good catch!